### ISSUE TITLE
POST call expected response body type also set to request body type

### ISSUE THREAD

**Issue by peetjvv** [axios/axios#4132]
#### Describe the bug
When specifying an expected `response.data` type to the post method it also sets that expected type to the request body. Also, when not specifying a type, the type for the `response.data` gets inferred to be the same as the type of the request body. This leads to typescript errors in VS Code and during compile time when the response and request bodies in code doesn't have the same type (which is usually the case). This was not an issue in v0.21.4 - which I reverted to in order to get around this issue.

#### To Reproduce
 1. Install axios v0.22.0
 2. Add a type for the `response.data` like in this example:
```tsx
axios.post<ExpectedResponseBody>(`post url`, { foo: "bar" }, {});
```
Also look at the CodeSandbox example [here](https://codesandbox.io/s/axios-v0-22-0-type-issue-joty3?file=/src/App.tsx:897-1031)

#### Expected behavior
When specifying the expected type of the `response.data` in the post call, the request body shouldn't have the same expected type. And the inferred type for the request body shouldn't automatically get applied to `response.data`.

#### Environment
 - Axios Version 0.22.0
 - Adapter N/A
 - Browser N/A
 - Browser Version N/A
 - Node.js Version: 14.17.3
 - OS: Win 10
 - Additional Library Versions: Typescript 4.4.3

#### Additional context/Screenshots
Type for post method in axios v0.21.4
![image](https://user-images.githubusercontent.com/22648295/135825306-2f1850cf-4200-4766-899a-057a98b4fe3c.png)

Type for post method in axios v0.22.0. Note that `T` is also set to request data and config in this version - which is what I believe is the cause of the issue.
![image](https://user-images.githubusercontent.com/22648295/135825453-d7abbb3a-4db1-4fd3-98a0-b25567c633a0.png)

**Comment by alessandrojcm** [comment]
Same problem here when updating to 0.22; it literally broke the whole app.

**Comment by TheVaan** [comment]
Duplicate to #4109

**Comment by peetjvv** [comment]
> 
> 
> Duplicate to #4109

Thank you. I couldn't find an existing issue for this.

**Comment by TheVaan** [comment]
You're welcome. Axios team messed up with the last release. Maybe they deserve issue overflow 😶

**Comment by tjhorner** [comment]
> Axios team messed up with the last release. Maybe they deserve issue overflow 😶

Yeah it's frankly concerning how this got through. We all make mistakes but this is a big one to miss 🙃 

It looks like their TypeScript tests may be incorrectly written? I'm not sure, and I don't really want to dig into it... but I'm not sure why they didn't catch this.

https://github.com/axios/axios/blob/76f09afc03fbcf392d31ce88448246bcd4f91f8c/test/typescript/axios.ts#L200-L202
